### ISSUE TITLE
attachContext

### ISSUE THREAD

**Issue by chatcrawler** [dami82/mutSignatures#16]
Hi dami82.

Thanks for your nice tool. I have installed and run the latest version from GitHub using RStudio v2022.07.1 Build 554 (R version 4.2.1).
#devtools::install_github("dami82/mutSignatures", force = TRUE, build_vignettes = TRUE)
#library(mutSignatures)

I have encountered an error at the step: 
#attachContext()

While I have no trouble getting through this step using your dataset and BSGenome for the human genome:
#data("mutSigData")

I always get an error using my own data, both VCF and my custom BSGenome object are successfully loaded.

The VCF files are read into R correctly using:
#importVCFfiles()

And the expected data frame with the expected 8 columns are generated in R:
#CHROM	POS	ID	REF	ALT	QUAL	FILTER	INFO

However, when I run 

attachContext(
mutData = x,
BSGenomeDb = my.genome,
chr_colName = "CHROM",
start_colName = "POS",
end_colName = "POS",
nucl_contextN = 3,
context_colName = "context",
skip_seqName_check = T
)

I get the following error message:

0 rows were excluded from analysis
Removing out-of-bounds positions...  428 records were removed.
An error has occurred!
Warning message:
In .Seqinfo.mergexy(x, y) :
  The 2 combined objects have no sequence levels in common. (Use
  suppressWarnings() to suppress this warning.)

This means that all 428 mutations in my data frame have been removed because they are deemed "out-of-bounds" for some reason. I guess it has to do with my BSGenome object, but it appears to be built correctly when following the *BSGenome guideline*:
http://bioconductor.org/packages/release/bioc/vignettes/BSgenome/inst/doc/BSgenomeForge.pdf

I am not certain what to do about it. Maybe you have a hint at what might have gone wrong at this step.
Thanks for your feedback.

Best

Seb

**Comment by dami82** [comment]
Hello Seb,

I just updated (8h ago) the gitHub attachContext() function and fixed a glitch in the code (additionally, I think I recognize the code you included in the message ^_^). Please, try the following steps:
1) restart R and (force) reinstall the package from GitHub
2) check the seqNames of your VCF file and your custom BSgenome object. They should match
3) if you still get errors, ask Clement to add you to our email thread ^_^ (or reply here if I got that wrong)

Best regards,
Damiano

——
Damiano Fantini

**Comment by chatcrawler** [comment]
Hello Damiano.

Thanks so much for your rapid reply, and help. The short version is: it worked, thanks a bunch!

The more verbose version of my answer: 
the update of your function was very timely, I updated the package from GitHub, and following your advice to restart R (which was also crucial), and now the attachContext() did work like a charm after that! Thanks a lot.

As a remark, I set "skip_seqName_check = T" in attachContext() because the error message was more informative (as you saw above), else I only got the message "An error occurred", which didn't give my a good idea about what went "wrong", here the program removed all data as it deemed them out-of-bounds, which lead to the error.

Great tool, keep it up and running!

Cheers,

Seb
